### ISSUE TITLE
Support selecting subset of properties on an expanded property

### ISSUE THREAD

**Issue by techniq** [voronov-maxim/OdataToEntity#17]
Currently the following fails: `Accounts(1)?$expand=Organization($select=Name)` (as well as `Accounts?$top=5&$expand=Organization($select=Name)`)

```
System.ArgumentException: Object of type 'System.Tuple`1[System.String]' cannot be converted to type 'Common.Models.Organizations.Taxonomy.Organization'.
   at System.RuntimeType.TryChangeType(Object value, Binder binder, CultureInfo culture, Boolean needsSpecialCast)
   at System.Reflection.MethodBase.CheckArguments(Object[] parameters, Binder binder, BindingFlags invokeAttr, CultureInfo culture, Signature sig)
   at System.Reflection.RuntimeMethodInfo.InvokeArgumentsCheck(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.SetValue(Object obj, Object value, BindingFlags invokeAttr, Binder binder, Object[] index, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.SetValue(Object obj, Object value, Object[] index)
   at OdataToEntity.AspNetCore.OeEntityAsyncEnumerator`1.<MoveNext>d__11.MoveNext()
```

**Comment by voronov-maxim** [comment]
This [test](https://github.com/voronov-maxim/OdataToEntity/blob/88d002bf85621e0287354c55dc3b6a38b83f0683/test/OdataToEntity.Test/Common/SelectTest.cs#L412) is similar to your error?

**Comment by techniq** [comment]
Odd, I went to bed last night, woke up, and could not reproduce.  Even my Chrome history shows I wasn't dreaming it 😄 

![image](https://user-images.githubusercontent.com/177476/40886727-6e2f0344-670b-11e8-9c48-175ccd54f7d6.png)

At first I thought it might be due to only selecting a single property on the expanded item (unlike your test you linked), but this does not appear to be the case either.

I pulled the source down last night to research #16 so maybe pulling some of your later changes also fixed this.

Thanks again.  I might have a few more bug reports coming your way after I research them more (still digging into #16 on and off).
